wasi/src/http_client.rs

```diff
@@ -0,0 +1,200 @@
+use anyhow::{anyhow, Context};
+use bytes::{BufMut, Bytes, BytesMut};
+use http::header::{HeaderName, HeaderValue};
+#[cfg(feature = "std")]
+use std::ops::Deref;
```

Comment:
This could be changed to `core::ops::Deref` to avoid the need to make it conditional on "std".